internal/io/csv.go

```diff
@@ -98,6 +101,40 @@ func ReadCSV(reader io.Reader, conf CSVConfig) (map[string]interface{}, []string
 		}
 	}
 
+	if conf.MissingColumnNameAlias != "" {
+		for i, name := range headers {
+			if name == "" {
+				headers[i] = conf.MissingColumnNameAlias
+			}
+		}
+	}
+
+	if conf.AddColumnIndexToDuplicateColumns {
+		headersMap := make(map[string]int)
```

Comment:
Consider making a new function for this code block taking headers and returning new headers since this is now becoming a quite significant piece of code.